types/params.go

```
@@ -43,3 +44,19 @@ func (pv *ParamsVersions) GetParamsForBTCHeight(btcHeight int32) (*GlobalParams,
 	}
 	return nil, fmt.Errorf("no version corresponds to the provided BTC height")
 }
+
+func (gp *GlobalParams) IsTimeBasedCap() (bool, error) {
+	if gp.StakingCap != 0 {
+		if gp.CapHeight != 0 {
+			return false, fmt.Errorf("only either of staking cap and cap height can be set")
```

We should probably panic here, as we are essentially running with invalid config. The only things we can do with this error later is to panic eitherway.

Also I would avoid nested ifs, as those are hard to read. Something like:

```
if gp.StakingCap != 0 && gp.CapHeight != 0 {
panic("only either of staking cap and cap height can be set")
}

if gp.StakingCap == 0 && gp.CapHeight == 0 {
panic("either of staking cap and cap height must be set")
}

return gp.CapHeight != 0
```

conveys the intend in much streigtforward fashion

---

Nice point! Fixed in [fb31a70](https://github.com/babylonchain/staking-indexer/pull/112/commits/fb31a70d4fa1bba5e6e5d297a2d2daba1c3040f1)